service/src/main/java/bio/terra/workspace/amalgam/landingzone/azure/LandingZoneApiDispatch.java

```diff
@@ -57,13 +60,23 @@ public ApiAzureLandingZoneResult createAzureLandingZone(
                     new LandingZoneInvalidInputException(
                         "LandingZoneTarget is required when creating an Azure landing zone"));
 
+    final LandingZoneTarget target = MapperUtils.LandingZoneTargetMapper.from(apiLandingZoneTarget);
+
+    // TODO: this check could be removed once WSM maintains the LZ id in the cloud context.
```

Comment:
I think this is independent of workspace/cloud context since this code relates to LZ creation. We probably still want it to fail if a LZ is already deployed in a given billing-profile (at least while we're enforcing a 1:1 relationship between LZ and billing profile).